### DISCUSSION TITLE
How to access dynamic variables in expression

### DISCUSSION THREAD

**Discussion by dnyaneshwarjadhav** [mikefarah/yq#1252] [Q&A]
How to use dynamic variables in expression

myres.txt
[
  {
    "name": "test-001",
    "is_preconfigured": true,
  },
  {
    "name": "my-connector",
    "is_preconfigured": 3
  },
  {
    "name": "checkerfrom_default2_002",
    "is_preconfigured": false,
  }
]

When i use direct command
yq e '. | .[].name | select(. == "my-connector")' myres.txt
This gives me result as _my-connector_

But when I use following way
NAME_IN_YAML="my-connector"
```
EXISTING_NAME=$(yq e '. | .[].name | select(. == $NAME_IN_YAML)' myres.txt)
```
This is blank result

**Comment by mikefarah** [comment]
You need to use the `env` operator https://mikefarah.gitbook.io/yq/operators/env-variable-operators#read-string-environment-variable

```
EXISTING_NAME=$(yq e '. | .[].name | select(. == strenv(NAME_IN_YAML))' myres.txt)
```
